src/index.ts

```diff
@@ -92,106 +92,85 @@ function getCommandDecoratorFields<T extends Command>(target: T, allDecoratorDat
  * @param target Command instance
  * @param options CLI arguments which was mapped to this map object by yargs
  */
-function initCommandFields<T extends Command>(target: T, options: { [key: string]: unknown }) {
+function initCommandFields<T extends LeafCommand>(target: T, options: Record<string, unknown>) {
   // eslint-disable-next-line guard-for-in
-  for (const key in target) {
-    const option = getOption(target, key)
+  for (const field in target) {
+    const key = findKey(target, field)
 
-    if (option) {
-      const value = options[option.key]
-
-      if (value === undefined) {
-        continue
-      }
-      target[key] = value as T[Extract<keyof T, string>]
-      continue
-    }
-
-    const externalOptionKey = getExternalOption(target, key)
-
-    if (externalOptionKey) {
-      const value = options[externalOptionKey]
-
-      if (value === undefined) {
-        continue
-      }
-      target[key] = value as T[Extract<keyof T, string>]
-      continue
-    }
-    const argument = getArgument(target, key)
-
-    if (argument) {
-      const value = options[argument.key]
-
-      if (value === undefined) {
-        continue
-      }
-      target[key] = value as T[Extract<keyof T, string>]
-      continue
+    if (key) {
+      Reflect.set(target, field, options[key])
     }
   }
 }
 
+function findKey<T extends LeafCommand, K extends Extract<keyof T, string>>(target: T, key: K): string | null {
+  return getOption(target, key)?.key || getExternalOption(target, key) || getArgument(target, key)?.key
+}
+
 class CommandBuilder {
   /**
    * Used for tests as well
    */
   public initedCommands: InitedCommand[]
   public runnable?: LeafCommand
-  public parser: Argv.Parser
-  public context!: Argv.Context | string
+  public parser: Madlad.Parser
+  public context!: Madlad.Context | string | { exitReason: string }
```

Comment:
I'm wondering why the `string | { exitReason: string }` is not part of `Madlad.Context` type if those types of values can come from the context?

Reply:
https://github.com/furious-commander/madlad/issues/3